tests/test_search.py

```diff
@@ -1,67 +1,200 @@
-                            }
-                        }]
+                    "match": {
+                        "files.sample_json.donor.species": "xxx"
                     }
                 }
             }
-        self.assertPostResponse(
-            '/v1/search',
-            json_request_body=(query),
-            expected_code=requests.codes.ok)
+
+        self.populate_search_index(self.index_document, 1)
+        self.verify_search_results(query)
+
+    def test_search_returns_error_when_invalid_query_used(self):
+        # Some types of invalid queries are detected by Elasticsearch DSL
+        # and others by Elasticsearch itself, and the response codes differ.
+        invalid_query_data = [
+            (
+                {
+                    "query": {
+                        "mtch": {
+                            "SomethingInvalid": "xxx"
+                        }
+                    }
+                },
+                requests.codes.bad_request
+            ),
+            (
+                {
+                    "query": {
+                        "match": ["SomethingInvalid", "xxx"]
+                    }
+                },
+                requests.codes.internal_server_error
+            )
+        ]
+
+        self.populate_search_index(self.index_document, 1)
+        for query_data in invalid_query_data:
+            with self.subTest("Invalid Queries"):
+                self.assertPostResponse(
+                    "/v1/search",
+                    json_request_body=(query_data[0]),
+                    expected_code=query_data[1])
+
+    def test_search_returns_X_results_when_X_documents_match_query(self):
+        test_matches = [0, 1, 9, 10, 11, 1000, 5000]
+        bundle_ids = []
+        indexed = 0
+        for x in test_matches:
+            create = x - indexed
+            indexed = x
+            bundle_ids.extend(self.populate_search_index(self.index_document, create))
+            with self.subTest("Search Returns %i Matches When %i Documents Indexed.".format(x, x)):
+                self.verify_search_results(self.smartseq2_paired_ends_query, x, bundle_ids)
+
+    def test_elasticsearch_exception(self):
+        # Test Elasticsearch exception handling by setting an invalid endpoint
+        es_logger = logging.getLogger("elasticsearch")
+        original_es_level = es_logger.getEffectiveLevel()
+        es_logger.setLevel("ERROR")
+        try:
+            os.environ['DSS_ES_ENDPOINT'] = "bogus"
+            response = self.assertPostResponse(
+                "/v1/search",
+                json_request_body=(self.smartseq2_paired_ends_query),
+                expected_code=requests.codes.internal_server_error)
+            self.assertEqual(response.json['code'], "elasticsearch_error")
+            self.assertEqual(response.json['title'], "Elasticsearch operation failed")
+        finally:
+            os.environ.pop('DSS_ES_ENDPOINT')
```

Comment:
this does not restore the state.  prefer that you save and restore that env var.